src/asm/lexer.c

```diff
@@ -1794,7 +1809,12 @@ static int yylex_NORMAL(void)
 		/* Handle strings */
 
 		case '"':
-			readString();
+			if (peekInternal(0) == '"' && peekInternal(1) == '"') {
```

Comment:
`peekInternal` should *never* be called outside of `peek`.
As for the code itself, what about this:
```
if peek(0) == '"'
    shiftChars(1)
    if peek(0) == '"'
        shiftChars(1)
        readString(true)
    else
        // Return empty string
    endif
else
    readString(false);
endc
```

This would save a `peek(1)`, which should be avoided whenever possible. (Arguably, that should be documented somewhere, possibly in a doc-comment above the function)

Reply:
It needs `peekInternal` so that if `S EQUS "\""` and `P EQUS "{S}" ; """`, then `P` will be `"\""`, not `""" ; """`. Likewise `P EQUS ""{S}` would be a syntax error, due to starting another string literal right after a `""` empty one, not the start of a triple-quoted string.

This may be a reason to avoid triple-quoted notation, but I would expect similar behavior with heredoc-style, `''single quotes''`, or whatever else, to avoid something like `S EQUS "<F"` and then `P EQUS <<{S}OO` being a valid way to start a multi-line string with identifier `FOO`.

That "what about this" code would need handling for the `else` cases in case `peek(0) != '"'`; reimplementing logic for all possible peeked characters would be hacky, but there's no `unpeek`/`rewind` facility (yet).

How about this: `yylex_NORMAL` just does `case '"': readString();` as before. Then after `readString` disables macros and interpolation, it can `peek(1)` and `peek(2)` to check for initial triple quotes before continuing with the actual string content.

Reply:
Then all expansions should be disabled as soon as the first quote is scanned.

Reply:
Yes, that's already the case in `master`.

Reply:
I mean, to scan for a triple quote. Due to that, too, I'd suggest moving this check within `readString`.

Reply:
Done. The triple-or-not logic is all in `readString` and only uses `peek`.

Reply:
Any reason for using `peek(1)` instead of what I suggested above?

Reply:
Your initial code block sample looks like a suggestion for different logic in `yylex_NORMAL`, outside the body of `readString`; and I thought "moving this check within `readString`" is what I did. Like I said, it could do `shiftChars(1); peek(0)` instead to check the potential third quote, but if that's *not* a third quote (e.g. `S EQUS ""<newline>` or `S EQUS "" ; empty`) then it would need to handle that specially.

Reply:
Not much more special handling than terminating the string, and `goto finish;`, right?
(On that note, is there any reason for the termination not to be handled at `finish`?)

Reply:
I was trying for a minimal edit PR, so left the `'\0'` termination as it is in `master`. You're right that it could/should be done once after `finish`.